From 23d9c4766ae2324cfbde15ae5cbebef00fd008b1 Mon Sep 17 00:00:00 2001
From: Kate <fukuro.s@yandex.ru>
Date: Fri, 21 Mar 2025 02:45:39 +0300
Subject: [PATCH 8/8] processing data from server by client

---
 20250317/1/client.py | 53 +++++++++++++++++++++++++++++++++++++-------
 20250317/1/server.py |  6 +++--
 2 files changed, 49 insertions(+), 10 deletions(-)

diff --git a/20250317/1/client.py b/20250317/1/client.py
index 31c502c..54ca358 100644
--- a/20250317/1/client.py
+++ b/20250317/1/client.py
@@ -5,9 +5,22 @@ import cowsay
 import readline
 import sys
 import socket
+from io import StringIO
 
 cows = cowsay.list_cows() + ['jgsbat']
 
+jgsbat = cowsay.read_dot_cow(StringIO(r"""
+    ,_                    _,
+    ) '-._  ,_    _,  _.-' (
+    )  _.-'.|\\\--//|.'-._  (
+     )'   .'\/o\/o\/'.   `(
+      ) .' . \====/ . '. (
+       )  / <<    >> \  (
+        '-._/``  ``\_.-'
+  jgs     __\\\'--'//__
+         (((""`  `"")))
+"""))
+
 def parse_args(args, param):
     args_parsed = {}
     for i in param:
@@ -64,15 +77,39 @@ class Client_MUD(cmd.Cmd):
         self.s = socket
         self.s.connect((self.host, self.port))
         return super().__init__(*args, **kwargs)
+    
+    def response_addmon(self, name, x, y, hello):
+        response = self.s.recv(1024).rstrip().decode()
+        print(f"Added monster {name} to ({x}, {y}) saying {hello}")
+        if response == '1': print("Replaced the old monster")
+
+    def response_attack(self, name):
+        response = self.s.recv(1024).rstrip().decode()
+        if response == 'no':
+            print(f"No {name} here")
+            return
+        damage, hp = [int(i) for i in response.split()]
+        print(f"Attacked {name}, damage {damage} hp")
+        if hp == 0:
+            print(f"{name} died")
+        else:
+            print(f"{name} now has {hp}")
 
-    def response(self):
-        return self.s.recv(1024).rstrip().decode()
+    def response_move(self):
+        response = self.s.recv(1024).rstrip().decode().split()
+        print(f"Moved to ({int(response[0])}, {int(response[1])})")
+        if len(response) > 2:
+            hello = ' '.join(response[3:])
+            if response[2] == 'jgsbat':
+                print(cowsay.cowsay(hello, cowfile=jgsbat))
+            else:
+                print(cowsay.cowsay(hello, cow=response[2]))
     
     def do_addmon(self, args):
         try:
             x, y, hp, hello, name = add_monster_check(args)
             self.s.sendall(f"addmon {name} {x} {y} {hp} {hello}\n".encode())
-            print(self.response())
+            self.response_addmon(name, x, y, hello)
         except Error as e:
             print(e.text)
 
@@ -80,7 +117,7 @@ class Client_MUD(cmd.Cmd):
         try:
             weapon, name = attack_check(args)
             self.s.sendall(f"attack {weapon} {name}\n".encode())
-            print(self.response())
+            self.response_attack(name)
         except Error as e:
             print(e.text)
     
@@ -89,28 +126,28 @@ class Client_MUD(cmd.Cmd):
             print(Error(1).text)
         else:
             self.s.sendall(f"move 0 -1\n".encode())
-            print(self.response())
+            self.response_move()
 
     def do_down(self, args):
         if len(args) != 0:
             print(Error(1).text)
         else:
             self.s.sendall(f"move 0 1\n".encode())
-            print(self.response())
+            self.response_move()
 
     def do_left(self, args):
         if len(args) != 0:
             print(Error(1).text)
         else:
             self.s.sendall(f"move -1 0\n".encode())
-            print(self.response())
+            self.response_move()
 
     def do_right(self, args):
         if len(args) != 0:
             print(Error(1).text)
         else:
             self.s.sendall(f"move 1 0\n".encode())
-            print(self.response())
+            self.response_move()
     
     def default(self, args):
         print("Invalid command")
diff --git a/20250317/1/server.py b/20250317/1/server.py
index 5bb421f..a2bdaf0 100644
--- a/20250317/1/server.py
+++ b/20250317/1/server.py
@@ -25,13 +25,14 @@ class Game:
 
     def encounter(self, x, y):
         if self.monsters[(x, y)]:
-            return f' {self.monsters[(x, y)].cow}'
+            return f' {self.monsters[(x, y)].cow} {self.monsters[(x, y)].phrase}'
         return ''
 
     def moving(self, player, d_x, d_y):
         s = player.move(d_x, d_y)
         if (player.x, player.y) in self.monsters:
             s += self.encounter(player.x, player.y)
+        print(s)
         return s
 
     def add_monster(self, x, y, hp, hello, name):
@@ -70,7 +71,8 @@ async def echo(reader, writer):
                 send = asyncio.create_task(reader.readline())
                 match request.result().decode().split():
                     case ['addmon', *args]:
-                        name, x, y, hp, hello = args
+                        name, x, y, hp = args[:4]
+                        hello = ' '.join(args[5:])
                         writer.write(game.add_monster(int(x), int(y), int(hp), hello, name).encode())
                     case ['attack', *args]:
                         x, y = player.x, player.y
-- 
2.43.0

